### PR TITLE
feat: add tournaments

### PR DESCRIPTION
**Pull request by ViktorMank**

### CODE REVIEW

File: src/components/tournaments/Modal/TournamentsModal.tsx [hunk 1]
```diff
@@ -1,75 +1,90 @@
...
-          />
-
-          <Paper p="xs" withBorder>
-            <Text>Каждый игрок играет в матчах до тех пор, пока не проиграет.</Text>
-          </Paper>
-        </Stack>
+        <Select
+          withAsterisk
+          label={t('modals.tournamentFormat')}
+          data={ModalSelectorsData.FORMATS}
```
Comment (remadyt) [inline src/components/tournaments/Modal/TournamentsModal.tsx:68]: incorrect constant name. renamte to TOURNOMENTS_CONFIG

File: src/components/tournaments/Modal/TournamentsModal.tsx [hunk 2]
```diff
@@ -1,75 +1,90 @@
...
+        <DateTimePicker
+          withAsterisk
+          valueFormat="ddd, MMM D, YYYY h:mm A"
+          label={t('modals.startTime')}
+          {...form.getInputProps('expectedDate')}
         />
 
         <Group position="right" mt="md">
-          <Button type="submit">{t('tournaments.createTournament')}</Button>
+          <Button type="submit">{t('modals.createTournament')}</Button>
```
Comment (remadyt) [inline src/components/tournaments/Modal/TournamentsModal.tsx:87]: why u rename to modals?

File: src/constants/databasePaths.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export enum DatabasePaths {
+  USERS = 'users',
+  TOURNAMENTS = 'tournaments',
```
Comment (remadyt) [inline src/constants/databasePaths.ts:3]: incorrect name, User, Tournaments for all enum use this style

File: src/helpers/emptyText.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { t } from 'i18next';
+
+const emptyText = (text?: string): string => {
+  return text || t('tournaments.unknownInfo');
```
Comment (remadyt) [inline src/helpers/emptyText.ts:4]: need to get rid of this feature

File: src/pages/TournamentInfo/TournamentInfo.tsx [hunk 5]
```diff
@@ -1,9 +1,118 @@
...
+import emptyText from 'helpers/emptyText';
+import { useTournaments } from 'store/store';
 
 const TournamentInfo: FC = () => {
-  return <Group>tournament info</Group>;
+  const { id } = useParams();
+  const { t } = useTranslation();
+  const { tournaments } = useTournaments((state) => state);
+
+  const tournamentData = useMemo(() => {
```
Comment (remadyt) [inline src/pages/TournamentInfo/TournamentInfo.tsx:23]: need getById request

File: src/store/slices/tournaments/tournamentsSlice.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    try {
+      setFirebaseData<ITournamentData>(data, DatabasePaths.TOURNAMENTS, data.id);
+
+      notifications.show({
+        title: t('notifications.successHeader'),
+        message: t('notifications.tournamentSuccess'),
+        color: 'green',
+      });
+    } catch (error) {
+      notifications.show({
```
Comment (remadyt) [inline src/store/slices/tournaments/tournamentsSlice.ts:51]: use common funcation for this

File: src/store/slices/tournaments/tournamentsSlice.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        title: t('notifications.errorHeader'),
+        message: t('notifications.errorCommon'),
+        color: 'red',
+      });
+    } finally {
+      set((state) => ({
+        ...state,
+        isFetching: false,
```
Comment (remadyt) [inline src/store/slices/tournaments/tournamentsSlice.ts:59]: check Andrey slices, now not needed make object copy

**Review by remadyt — CHANGES_REQUESTED** [review]

File: src/components/tournaments/Modal/config.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { csgoicon, dotaicon } from 'assets/images';
+
+export const ModalSelectorsData = {
```
Comment (suveiko) [inline src/components/tournaments/Modal/config.ts:3]: Can we create a type for this object?

File: src/store/slices/tournaments/tournamentsSlice.ts [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+
+export const tournamentsSlice: GenericStateCreator<TournamentsStore> = (set, get) => ({
+  ...get(),
+  isFetching: false,
+  currentPage: 1,
+  tournamentsPerPage: 10,
+  tournaments: [],
+
+  addTournament: async (data: ITournamentData) => {
+    set((state) => ({
```
Comment (thelastandrew) [inline src/store/slices/tournaments/tournamentsSlice.ts:37]: you can now do the same this way useing `immer`:
```
import { produce } from 'immer';

/// you code

set(produce(state: TournamentsStore) => { state.isFetching = true; });
```

**Review by thelastandrew — CHANGES_REQUESTED** [review]

**Review by remadyt — APPROVED** [review]

**Review by thelastandrew — APPROVED** [review]

**Pull request merged**